### DISCUSSION TITLE
Correlations between LLM requests

### DISCUSSION THREAD

**Discussion by code-junkie-403** [microsoft/semantic-kernel#7342] [General]
Hi, we are using a custom HTTP handler to track all LLM requests in each plugin. I'd like to check if there's a correlation ID we can use to link them together? Eg SK triggers Plugin1 -> Plugin2 -> Plugin3, we'd like to correlate all the LLM requests made by these plugins. 

Thanks

**Comment by dmytrostruk** [comment]
@code-junkie-403 If you are interested in .NET, you can use log scopes to attach correlation ID to each log that's created during the same operation:
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-8.0#log-scopes
https://github.com/microsoft/semantic-kernel/blob/6b01468efb81918c002a23e726cca1431401f855/dotnet/samples/Concepts/Filtering/TelemetryWithFilters.cs#L68-L79

If you need programmatic access to correlation ID, you can use traces and activities:
- https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-concepts
- https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-instrumentation-walkthroughs
